app/src/main/java/fr/free/nrw/commons/depictions/SubClass/SubDepictionListPresenter.java

```diff
@@ -0,0 +1,135 @@
+package fr.free.nrw.commons.depictions.SubClass;
+
+import java.lang.reflect.Proxy;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+
+import fr.free.nrw.commons.explore.depictions.DepictsClient;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearch;
+import fr.free.nrw.commons.explore.recentsearches.RecentSearchesDao;
+import fr.free.nrw.commons.upload.structure.depicts.DepictedItem;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.CompositeDisposable;
+import io.reactivex.schedulers.Schedulers;
+import timber.log.Timber;
+
+public class SubDepictionListPresenter implements SubDepictionListContract.UserActionListener {
+
+    private static final SubDepictionListContract.View DUMMY = (SubDepictionListContract.View) Proxy
+            .newProxyInstance(
+                    SubDepictionListContract.View.class.getClassLoader(),
+                    new Class[]{SubDepictionListContract.View.class},
+                    (proxy, method, methodArgs) -> null);
+
+    private  SubDepictionListContract.View view = DUMMY;
+    RecentSearchesDao recentSearchesDao;
+    String query;
+    protected CompositeDisposable compositeDisposable = new CompositeDisposable();
+    DepictsClient depictsClient;
+    private static int TIMEOUT_SECONDS = 15;
+    private List<DepictedItem> queryList = new ArrayList<>();
+
+    @Inject
+    public SubDepictionListPresenter(RecentSearchesDao recentSearchesDao, DepictsClient depictsClient) {
+        this.recentSearchesDao = recentSearchesDao;
+        this.depictsClient = depictsClient;
+    }
+    @Override
+    public void onAttachView(SubDepictionListContract.View view) {
+        this.view = view;
+    }
+
+    @Override
+    public void onDetachView() {
+        this.view = DUMMY;
+    }
+
+    @Override
+    public void saveQuery() {
+        RecentSearch recentSearch = recentSearchesDao.find(query);
+
+        // Newly searched query...
+        if (recentSearch == null) {
+            recentSearch = new RecentSearch(null, query, new Date());
+        } else {
+            recentSearch.setLastSearched(new Date());
+        }
+        recentSearchesDao.save(recentSearch);
+    }
+
+    @Override
+    public void fetchThumbnailForEntityId(String entityId, int position) {
+        compositeDisposable.add(depictsClient.getP18ForItem(entityId)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .subscribe(response -> {
+                    Timber.e("line67" + response);
```

Comment:
Is this intended to be permanent? Maybe make it debug level :-)

Reply:
no this shouldn't be permanent. I think it will be better to remove it